ConverterViewController.swift

```diff
@@ -0,0 +1,174 @@
+                    
+                    self._requestedWasOk = true
+                    
+                    self._exchangesDictionary = convertedJsonIntoDict["rates"] as? [String : Double]
+                    
+                    completion() //callback
+                    
+                }
+            } catch let error as NSError {
+                print(error.localizedDescription)
+                
+                self._requestedWasOk = false
+            }
+            
+        }
+        
+        task.resume()
+    }
+    
+    //displays equivalent values in different format currencies.
+    func updateValues() {
+     
+      dispatch_async(dispatch_get_main_queue(), {
+        
+            //Calculates and shows exchange value for UK Pounds
+            if var priceGBP = self._exchangesDictionary![CURRENCIES["UK"]!] {
+                priceGBP = priceGBP * self._dollarAmount!;
+                self._currencyFormatter.currencyCode = CURRENCIES["UK"]!
+                self.lblGBP.text = self._currencyFormatter.stringFromNumber(priceGBP);
+            }
+        
+            //Calculates and shows exchange value for Euros
+            if var priceEUR = self._exchangesDictionary![CURRENCIES["EUR"]!] {
+                priceEUR = priceEUR * self._dollarAmount!;
+                self._currencyFormatter.currencyCode = CURRENCIES["EUR"]!
+                self.lblEUR.text = self._currencyFormatter.stringFromNumber(priceEUR);
+                
+            }
+        
+            //Calculates and shows exchange value for Brasilian reals
+            if var priceBRL = self._exchangesDictionary![CURRENCIES["BRA"]!] {
+                priceBRL = priceBRL * self._dollarAmount!;
+                self._currencyFormatter.currencyCode = CURRENCIES["BRA"]!
+                self.lblBRL.text = self._currencyFormatter.stringFromNumber(priceBRL);
+            }
+        
+            //Calculates and shows exchange value for Japan Yens
+            if var priceJPY = self._exchangesDictionary![CURRENCIES["JAP"]!] {
+                priceJPY = priceJPY * self._dollarAmount!;
+                self._currencyFormatter.currencyCode = CURRENCIES["JAP"]!
+                self.lblJPY.text = self._currencyFormatter.stringFromNumber(priceJPY);
+            }
+      })
+    
+    }
+    
+    // detects when dollarNumberTextField changed
+    @IBAction func didDollarChange(sender: AnyObject) {
+        
+        self._dollarAmount = DEFAULTAMOUNT //just in case the textfield format is not valid
+        
+        if let dollarTextNumer: String = self.dollarNumberTextField.text {
+            if let dollarConverted = Double(dollarTextNumer) {
+                self._dollarAmount = dollarConverted
+            }
+        }
+        
+        self.updateValues()
+    }
+    
+//MARK: Textfield methods 
+    
+    //Sets a max value for dollarNumberTextField
+    func textField(textField: UITextField, shouldChangeCharactersInRange range: NSRange,
```

Comment:
Swift allows you to do something nice to structure your code with `extensions`. 
You would usually make an extension block for each protocol you are conforming to, with it's implementation.
For example.

```
class ConverterViewController { 

let tableView = UITableView()
let textField = UITextField()

 // All of your VC's logic goes here.

tableView.delegate = self
tableView.datasource = self
textField.delegate = self
}
```
Now you want to conform to the TableView's protocols.
```
extension ConverterViewController: UITableViewDelegate, UITableViewDatasource {
// Implement the methods you need.
}
```

Now for the UITextField:
```
extension ConverterViewController: UITextFieldDelegate {
// The method you want to implement.
}
```

This allows your code to be more organized into logic pieces, making it a lot more readable.